pkg/plugins/backendplugin/grpcplugin/client_v2.go

```diff
@@ -338,3 +347,35 @@ func (c *ClientV2) ConvertObjects(ctx context.Context, req *backend.ConversionRe
 
 	return backend.FromProto().ConversionResponse(protoResp), nil
 }
+
+// handleGrpcStatusError handles gRPC status errors and sets the error source based on the error metadata.
+func handleGrpcStatusError(ctx context.Context, err error) error {
+	st := status.Convert(err)
+	for _, detail := range st.Details() {
+		switch t := detail.(type) {
+		case *errdetails.ErrorInfo:
```

Comment:
Since this type switch only has one case, can we replace it with this instead?

```go
		if detail, ok := detail.(*errdetails.ErrorInfo); ok {
			// use `detail.Metadata[errorSourceMetadataKey]` ...
		}
```